flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/GenericInMemoryCatalog.java

```diff
@@ -531,11 +520,18 @@ public boolean partitionExists(ObjectPath tablePath, CatalogPartitionSpec partit
 		return partitions.containsKey(tablePath) && partitions.get(tablePath).containsKey(partitionSpec);
 	}
 
+	private void ensureFullPartitionSpec(ObjectPath tablePath, CatalogPartitionSpec partitionSpec)
```

Comment:
TableNotExistException doesn't seem being thrown anywhere from this method.

Reply:
It's thrown from getTable(), but that doesn't seem optimal. I'll make isFullPartitionSpec throw TableNotExistException.